### PR TITLE
Update quoted identifiers BBE

### PR DESCRIPTION
**Pull request by HindujaB**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

**Comment by niveathika** [comment]
@HindujaB Sync it with master and ensure the build passes

**Comment by Kishanthan** [comment]
@HindujaB , shall we also include function names, type names, etc having quoted identifier support?

File: examples/quoted-identifiers/quoted_identifiers.bal [hunk 1]
```diff
@@ -19,4 +19,17 @@ public function main() {
     // Define a variable where the variable name starts with a digit.
     int '1PlusI = 1 + i;
     io:println('1PlusI);
+
+    // Define a variable where the variable name contains special characters with a preceding `\`.
```
Comment (praneesha) [inline examples/quoted-identifiers/quoted_identifiers.bal:23]:
```suggestion
    // Define a variable of which the name contains special characters with a preceding `\`.
```

File: examples/quoted-identifiers/quoted_identifiers.bal [hunk 2]
```diff
@@ -19,4 +19,17 @@ public function main() {
...
+    int '\{add\#5\} = 5 + i;
+    io:println('\{add\#5\});
+
+    // Define a variable where the variable name contains unicode characters.
```
Comment (praneesha) [inline examples/quoted-identifiers/quoted_identifiers.bal:27]:
```suggestion
    // Define a variable of which the variable name contains unicode characters.
```

File: examples/quoted-identifiers/quoted_identifiers.bal [hunk 3]
```diff
@@ -19,4 +19,17 @@ public function main() {
...
+    string 'üňĩćőđę_ňāɱȇ = "John doe";
+    io:println('üňĩćőđę_ňāɱȇ);
+
+    // Define a variable where the variable name contains unicode character specified by hexadecimal code points.
```
Comment (praneesha) [inline examples/quoted-identifiers/quoted_identifiers.bal:31]:
```suggestion
    // Define a variable of which the variable name contains unicode characters specified by hexadecimal code points.
```

File: examples/quoted-identifiers/quoted_identifiers.description [hunk 4]
```diff
@@ -1,3 +1,3 @@
 // Ballerina supports quoted identifiers, making it possible to use arbitrary non-empty strings as identifiers,
```
Comment (praneesha) [inline examples/quoted-identifiers/quoted_identifiers.description:1]:
```suggestion
// Ballerina supports quoted identifiers making it possible to use arbitrary non-empty strings as identifiers
```

**Comment by HindujaB** [comment]
> @HindujaB , shall we also include function names, type names, etc having quoted identifier support?

@Kishanthan Added some more applications in the last commit

File: examples/quoted-identifiers/quoted_identifiers.bal [hunk 5]
```diff
@@ -19,4 +31,28 @@ public function main() {
     // Define a variable where the variable name starts with a digit.
```
Comment (praneesha) [inline examples/quoted-identifiers/quoted_identifiers.bal:31]:
```suggestion
    // Define a variable of which the variable name starts with a digit.
```

File: examples/quoted-identifiers/quoted_identifiers.bal [hunk 6]
```diff
@@ -19,4 +31,28 @@ public function main() {
...
+    // Define a variable of which the variable name contains unicode characters specified by hexadecimal code points.
+    string 'unicode_\u{2324} = "Jane doe";
+    io:println('unicode_\u{2324});
+
+    // Invoke the function named `get_ɱȇşşağę_for_\#` with the variable named `int`.
+    io:println('get_ɱȇşşağę_for_\#('int));
+
+    'Person_\{name\&Ȧɢέ\} person = {'first\ name: "Tom", 'Ȧɢέ:25};
+
+    // Member access can be used to access the members of type `'Person_{name&Ȧɢέ}` by referring the field name
```
Comment (praneesha) [inline examples/quoted-identifiers/quoted_identifiers.bal:52]:
```suggestion
    // Member access can be used to access the members of type `'Person_{name&Ȧɢέ}` by referring to the field name
```

File: examples/quoted-identifiers/quoted_identifiers.bal [hunk 7]
```diff
@@ -19,4 +31,28 @@ public function main() {
...
+    // without escaping the special characters using a `\`
```
Comment (praneesha) [inline examples/quoted-identifiers/quoted_identifiers.bal:53]:
```suggestion
    // without escaping the special characters using a `\`.
```

File: examples/quoted-identifiers/quoted_identifiers.bal [hunk 8]
```diff
@@ -19,4 +31,28 @@ public function main() {
...
+    io:println(person["first name"]);
+
+    // The required field `'Ȧɢέ` can be accessed using field based access
```
Comment (praneesha) [inline examples/quoted-identifiers/quoted_identifiers.bal:56]:
```suggestion
    // The required field `'Ȧɢέ` can be accessed using field based access.
```

File: examples/quoted-identifiers/quoted_identifiers.description [hunk 9]
```diff
@@ -1,3 +1,3 @@
-// Ballerina supports quoted identifiers, making it possible to use arbitrary non-empty strings as identifiers,
-// including those that are not generally allowed as identifiers (e.g., reserved keywords).
+// Ballerina supports quoted identifiers making it possible to use arbitrary non-empty strings as identifiers
```
Comment (praneesha) [inline examples/quoted-identifiers/quoted_identifiers.description:1]:
```suggestion
// Ballerina supports quoted identifiers making it possible to use arbitrary, non-empty strings as identifiers
```

File: examples/quoted-identifiers/quoted_identifiers.bal [hunk 10]
```diff
@@ -1,10 +1,22 @@
 import ballerina/io;
 
-// Use the reserved keyword `function` as the name of the function.
+// Use the reserved keyword `function` as the function identifier.
```
Comment (MaryamZi) [inline examples/quoted-identifiers/quoted_identifiers.bal:3]: Was there a particular reason to change this? IMO, it's clearer to say name.

Should we at least change this to
```suggestion
// Use the reserved keyword `function` as the identifier of the function.
```

File: examples/quoted-identifiers/quoted_identifiers.bal [hunk 11]
```diff
@@ -1,10 +1,22 @@
...
 function 'function(int val) returns int {
     return val + 1;
 }
 
+// Use different types of characters in the function identifier.
```
Comment (MaryamZi) [inline examples/quoted-identifiers/quoted_identifiers.bal:8]: Would "kinds" be better here?

```suggestion
// Use different kinds of characters in the identifier of the function.
```

File: examples/quoted-identifiers/quoted_identifiers.bal [hunk 12]
```diff
@@ -1,10 +1,22 @@
...
+function 'get_ɱȇşşağę_for_\#(int val) returns string {
+    return val.toString() + " is a numeric value" ;
+}
+
+// Use different types of characters in the type definition.
+// The quoted identifier syntax is used to refer to both the type name and the field names.
```
Comment (MaryamZi) [inline examples/quoted-identifiers/quoted_identifiers.bal:14]:
```suggestion
// The quoted identifier syntax is used in both the type name and the field names.
```

File: examples/quoted-identifiers/quoted_identifiers.bal [hunk 13]
```diff
@@ -5,18 +5,54 @@ function 'function(int val) returns int {
...
+|};
+
 public function main() {
 
     // Use the reserved keyword `int` as the name of the variable.
     int 'int = 1;
 
     // Invoke the function named `function` with the variable named `int`.
-    // The quoted identifier syntax is used to refer to both the function and
+    // The quoted identifier syntax is used in both the function and
```
Comment (MaryamZi) [inline examples/quoted-identifiers/quoted_identifiers.bal:26]: On second thoughts, I think what was originally there is correct. Sorry!

Shall we change this back to
```suggestion
    // The quoted identifier syntax is used to refer to both the function and
```

File: examples/quoted-identifiers/quoted_identifiers.bal [hunk 14]
```diff
@@ -5,6 +5,18 @@ function 'function(int val) returns int {
     return val + 1;
 }
 
+// Use different kinds of characters in the name of the function.
+function 'get_ɱȇşşağę_for_\#(int val) returns string {
+    return val.toString() + " is a numeric value" ;
+}
+
+// Use different kinds of characters in the type definition.
+// The quoted identifier syntax is used to refer to both the type name and the field names.
```
Comment (MaryamZi) [inline examples/quoted-identifiers/quoted_identifiers.bal:14]:
```suggestion
// The quoted identifier syntax is used in both the type name and the field names.
```

**Review by MaryamZi — APPROVED** [review]

**Review by praneesha — APPROVED** [review]

**Pull request merged**